### PR TITLE
fix:  --install-if-missing when using version alias

### PR DESCRIPTION
**Pull request by AlexMunoz**
I am trying to fix this issue https://github.com/Schniz/fnm/issues/482 based on my basic Rust knowledge. Any advice  to improve this further would be appreciated. If I am completely wrong, please let me know that as well.

### CODE REVIEW

File: src/commands/use.rs [hunk 1]
```diff
@@ -19,6 +19,29 @@ pub struct Use {
...
+
+        Self {
+            version: Some(UserVersionReader::Direct(requested_version)),
+            install_if_missing,
+        }
+        .apply(config)?;
+
+        return Ok(());
+    }
+}
```
Comment (Schniz) [inline src/commands/use.rs:43]: can we move this out of the `impl Use`?

Dangling functions are okay -- `fn install_new_version` is all good. We can also move the `ensure` block inside to share it among the two branches.

**Review by Schniz — COMMENTED** [review]
YES! Very good! Thank you.
Can you add a test or at least verify that it works?
Also please look at my comments 🙏

**Comment by AlexMunoz** [comment]
Hi @Schniz  I have made some changes and also added some tests!

**Comment by Schniz** [comment]
Thanks! Eager to see. But I'm afk this weekend. Will check on Sunday! 😃😃😃

File: src/commands/use.rs [hunk 2]
```diff
@@ -61,26 +60,9 @@ impl Command for Use {
...
-                    .context(InstallError)?;
-
-                    Self {
-                        version: Some(UserVersionReader::Direct(requested_version)),
-                        install_if_missing: self.install_if_missing,
-                    }
-                    .apply(config)?;
-
+                    let can_install =
+                        self.install_if_missing || should_install_interactively(&requested_version);
```
Comment (Schniz) [inline src/commands/use.rs:64]: can we move the `should_install_interactively` to be a part of the extracted function?
this is for suggesting the user `hey, this is missing, wanna install this?` -- and not just waiting for the `--install-if-missing` flag :)

Comment (Schniz) [reply src/commands/use.rs:64]: Other than that, it looks reaaaaaaally good. I hope it was easy figuring out how to write the e2e tests :)

Comment (AlexMunoz) [reply src/commands/use.rs]: I made the change ;)  Yes, it took me a while to figure out how to create the snapshots. :D

**Review by Schniz — APPROVED** [review]
Amazing!

**Comment by AlexMunoz** [comment]
don't forget the PR label :smile:

**Pull request merged**